torch/csrc/jit/codegen/cuda/ir_interface_nodes.h

```diff
@@ -79,12 +86,26 @@ class TORCH_CUDA_CU_API Double : public Val {
     return maybe_value_;
   }
 
-  bool sameAs(const Statement* other) const override;
+  bool sameAs(const Statement* other) const override {
+    if (this == other) {
+      return true;
+    }
+    if (!other->isA<Double>()) {
```

Comment:
I think this should be
```suggestion
    if (!other->isA<FloatingPoint>()) {
```

Reply:
Thanks!